bears/vcs/git/GitCommitBear.py

```diff
@@ -110,6 +114,38 @@ def check_shortlog(self,
                     self,
                     "Shortlog of HEAD commit does not match given regex.")
 
+        if shortlog_imperative_check:
+            colon_pos = shortlog.find(':')
+            shortlog = shortlog[colon_pos:] if colon_pos != -1 else shortlog
+            has_flaws = self.check_imperative(shortlog)
+            if has_flaws:
+                bad_words = ', '.join([words[0] for words in has_flaws])
```

Comment:
Is it fast? Wasn't aware of that. I'll look into that. Thanks.
ᐧ

On Mon, Apr 4, 2016 at 11:29 PM, Mischa Krüger <notifications@github.com>
wrote:

> In bears/vcs/git/GitCommitBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/315#discussion_r58418718>
> :
>
> > @@ -110,6 +114,38 @@ def check_shortlog(self,
> >                      self,
> >                      "Shortlog of HEAD commit does not match given regex.")
> >
> > +        if shortlog_imperative_check:
> > +            colon_pos = shortlog.find(':')
> > +            shortlog = shortlog[colon_pos:] if colon_pos != -1 else shortlog
> > +            has_flaws = self.check_imperative(shortlog)
> > +            if has_flaws:
> > +                bad_words = ', '.join([words[0] for words in has_flaws])
>
> if you want it faster:
> you can use map and operator.itemgetter, map tries to vectorize the
> calculations then.
>
> >>> from operator import itemgetter
> >>> rows = [(1, 2), (3, 4), (5, 6)]
> >>> map(itemgetter(1), rows)
> [2, 4, 6]
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/315/files/23d9af66bcb498b19adffe8effb5af1ec8aa4f40#r58418718>
>